src/state/actions/scigateway.actions.test.tsx

```diff
@@ -542,6 +542,52 @@ describe('scigateway actions', () => {
     expect(actions).toContainEqual(siteLoadingUpdate(false));
   });
 
+  it("dispatches a site loading update but doesn't wait for register route event if it's already been recieved", async () => {
+    (mockAxios.get as jest.Mock).mockImplementation(() =>
+      Promise.resolve({
+        data: {
+          'ui-strings': 'res/default.json',
+        },
+      })
+    );
+
+    const asyncAction = configureSite();
+    const actions: Action[] = [];
+    const dispatch = (action: Action): void | Promise<void> => {
+      if (typeof action === 'function') {
+        action(dispatch);
+        return Promise.resolve();
+      } else {
+        actions.push(action);
+      }
+    };
+
+    const state = JSON.parse(JSON.stringify(initialState));
+    state.authorisation.provider = new TestAuthProvider('token');
+    state.plugins = [
+      {
+        section: 'Analysis',
+        link: '/test',
+        plugin: 'demo_plugin',
+        displayName: 'Demo Plugin Analysis',
+      },
+    ];
+    const getState = (): Partial<StateType> => ({
+      scigateway: state,
+      router: {
+        location: { ...createLocation('/test'), query: {} },
+        action: 'PUSH',
+      },
+    });
+
+    await asyncAction(dispatch, getState);
+
+    const eventListenerSpy = jest.spyOn(document, 'addEventListener');
+
+    expect(eventListenerSpy).not.toHaveBeenCalled();
```

Comment:
Doesn't a spy need to be initialised before the tested function is called? Otherwise it would return true for not called, right? Granted, this is an async function so maybe you'd get mixed results